In [1]:
import pandas as pd 

DATA_PATH = "../data" 
df = pd.read_csv(f"{DATA_PATH}/lunar.csv")

df.head()

,Catalog Number,Calendar Date,Eclipse Time,Delta T (s),Lunation Number,Saros Number,Eclipse Type,Quincena Solar Eclipse,Gamma,Penumbral Magnitude,Umbral Magnitude,Latitude,Longitude,Penumbral Eclipse Duration (m),Partial Eclipse Duration (m),Total Eclipse Duration (m)
0,1,-1999 June 26,14:13:28,46437,-49456,17,N,t-,-1.0981,0.8791,-0.1922,24S,22W,268.8,-,-
1,2,-1999 November 21,20:23:49,46427,-49451,-16,N,-a,-1.1155,0.8143,-0.1921,15N,98W,233.4,-,-
2,3,-1998 May 17,05:47:36,46416,-49445,-11,P,-t,0.8988,1.2105,0.2069,13S,89E,281.7,102.7,-
3,4,-1998 November 11,05:15:58,46404,-49439,-6,P,-a,-0.4644,2.0382,0.9740,12N,113E,343.4,200.8,-
4,5,-1997 May 6,18:57:01,46392,-49433,-1,T+,pp,0.1003,2.6513,1.6963,11S,92W,322.8,213.5,98.2


In [8]:
df.shape

(12064, 16)

In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 12064 entries, 0 to 12063
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Catalog Number                  12064 non-null  int64  
 1   Calendar Date                   12064 non-null  str    
 2   Eclipse Time                    12064 non-null  str    
 3   Delta T (s)                     12064 non-null  int64  
 4   Lunation Number                 12064 non-null  int64  
 5   Saros Number                    12064 non-null  int64  
 6   Eclipse Type                    12064 non-null  str    
 7   Quincena Solar Eclipse          12064 non-null  str    
 8   Gamma                           12064 non-null  float64
 9   Penumbral Magnitude             12064 non-null  float64
 10  Umbral Magnitude                12064 non-null  float64
 11  Latitude                        12064 non-null  str    
 12  Longitude                       12064 non-n

In [ ]:
# Check for missing values 
df.isnull().sum() 

Catalog Number                    0
Calendar Date                     0
Eclipse Time                      0
Delta T (s)                       0
Lunation Number                   0
Saros Number                      0
Eclipse Type                      0
Quincena Solar Eclipse            0
Gamma                             0
Penumbral Magnitude               0
Umbral Magnitude                  0
Latitude                          0
Longitude                         0
Penumbral Eclipse Duration (m)    0
Partial Eclipse Duration (m)      0
Total Eclipse Duration (m)        0
dtype: int64

In [11]:
df.columns.tolist()


['Catalog Number',
 'Calendar Date',
 'Eclipse Time',
 'Delta T (s)',
 'Lunation Number',
 'Saros Number',
 'Eclipse Type',
 'Quincena Solar Eclipse',
 'Gamma',
 'Penumbral Magnitude',
 'Umbral Magnitude',
 'Latitude',
 'Longitude',
 'Penumbral Eclipse Duration (m)',
 'Partial Eclipse Duration (m)',
 'Total Eclipse Duration (m)']

In [12]:
df.describe()

,Catalog Number,Delta T (s),Lunation Number,Saros Number,Gamma,Penumbral Magnitude,Umbral Magnitude,Penumbral Eclipse Duration (m)
count,12064.000000,12064.000000,12064.000000,12064.000000,12064.000000,12064.000000,12064.000000,12064.000000
mean,6032.500000,12116.476044,-18531.392822,80.505056,0.002490,1.418656,0.400175,269.971941
std,3482.721158,13584.785584,17887.218741,48.416007,0.910505,0.832263,0.832954,79.945444
min,1.000000,-6.000000,-49456.000000,-20.000000,-1.582700,0.000400,-1.095800,5.200000
25%,3016.750000,962.000000,-33923.250000,40.000000,-0.788825,0.684425,-0.334000,223.100000
50%,6032.500000,5597.000000,-18445.500000,80.000000,0.001750,1.417450,0.400450,295.000000
75%,9048.250000,20901.500000,-3067.750000,121.000000,0.791725,2.136900,1.117925,327.800000
max,12064.000000,46437.000000,12378.000000,183.000000,1.579100,2.908900,1.882100,379.500000


## Summary
The lunar eclipse dataset contains 12,064 records with 16 columns, with no missing 
values. It includes multiple duration fields (penumbral, partial, total) since lunar 
eclipses pass through several phases. Key columns for the dashboard: Calendar Date, 
Eclipse Type, Latitude/Longitude, and Umbral Magnitude.